compiler/rustc_session/src/config.rs

```diff
@@ -1023,34 +1023,28 @@ pub fn to_crate_config(cfg: FxHashSet<(String, Option<String>)>) -> CrateConfig
 
 /// The parsed `--check-cfg` options
 pub struct CheckCfg<T = String> {
-    /// Set if `names()` checking is enabled
-    pub names_checked: bool,
-    /// The union of all `names()`
-    pub names_valid: FxHashSet<T>,
-    /// The set of names for which `values()` was used
-    pub values_checked: FxHashSet<T>,
-    /// The set of all (name, value) pairs passed in `values()`
-    pub values_valid: FxHashSet<(T, T)>,
+    /// The set of all `names()`, if none no names checking is performed
+    pub names_valid: Option<FxHashSet<T>>,
+    /// The set of all `values()`, if none no values chcking is performed
+    pub values_valid: Option<FxHashMap<T, FxHashSet<T>>>,
```

Comment:
I don't think this is correct, for values there's no single `Option`.
`--check-cfg values(foo, ...)` enables value checking only for `foo` but not for other names.

Reply:
Ok, I see, if values are not checked for the name `bar`, then `bar` is simply not in the map, so it doesn't affect correctness.

But I still think the `Option` is redundant there, because there's no observable difference between `None` and `Some(<empty map>)`.

>It's also probably faster to check `None` than trying to get a element from an empty `FxHashMap`

https://github.com/rust-lang/hashbrown/pull/305

Reply:
> Ok, I see, if values are not checked for the name `bar`, then `bar` is simply not in the map, so it doesn't affect correctness.

Yes exactly, but being in it without wanting to be in would be a problem because not having values is a perfectly valid state, ex: `unix`, `windows`, ... this would affect correctness.

> But I still think the `Option` is redundant there, because there's no observable difference between `None` and `Some(<empty map>)`.

Fair enough, done in https://github.com/rust-lang/rust/pull/94175/commits/106bac126e126959311b9504ea08b80dc39c39eb.